cpp/src/slice2js/Gen.cpp

```diff
@@ -163,6 +163,27 @@ namespace
         }
     }
 
+    void writeDeprecated(Output& out, const CommentPtr& comment, const ContainedPtr& contained)
+    {
+        // JavaScript doesn't provide a way to deprecate elements other than by using a comment, so we map both the
+        // Slice @deprecated tag and the deprecated metadata argument to a `@deprecated` JSDoc tag.
+        if ((comment && comment->isDeprecated()) || contained->isDeprecated())
+        {
+            out << nl << " * @deprecated";
+            // If a reason was supplied, append it after the `@deprecated` tag. If no reason was supplied, fallback to
+            // the deprecated metadata argument.
+            if (!comment->deprecated().empty())
+            {
+                out << " ";
+                writeDocLines(out, comment->deprecated(), false);
+            }
+            else if (auto deprecated = contained->getDeprecationReason())
```

Comment:
We should think about and settle on a good terminology for deprecated. I am not a fan of deprecationReason when the metadata and doc-comment are 'deprecated'.